dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ClassLoaderRef.java

```diff
@@ -0,0 +1,160 @@
+        super(appClassLoader.getParent());
+        classLoaderWeakRef = new WeakReference<>(appClassLoader);
+    }
+
+    @Override
+    protected final Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
+        return loadClass(name, resolve, false, false);
+    }
+
+    @Override
+    public URL getResource(String resName) {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.getResource(resName) : parent.getResource(resName);
+    }
+
+    @Override
+    public InputStream getResourceAsStream(String resName) {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.getResourceAsStream(resName) : parent.getResourceAsStream(resName);
+    }
+
+    @Override
+    public Enumeration<URL> getResources(String resName) throws IOException {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.getResources(resName) : parent.getResources(resName);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof ClassLoaderRef) {
+            return Objects.equals(classLoaderWeakRef.get(), ((ClassLoaderRef) o).classLoaderWeakRef.get());
+        }
+        return false;
+    }
+
+    @Override
+    public boolean addTransformer(ClassFileTransformer cft) {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        if (appCL != null) {
+            return appCL.addTransformer(cft);
+        }
+        return false;
+    }
+
+    @Override
+    public ClassLoader getThrowawayClassLoader() {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.getThrowawayClassLoader() : parent;
+    }
+
+    @Override
+    public Class<?> publicDefineClass(String name, byte[] b, ProtectionDomain protectionDomain) {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        return appCL != null ? appCL.publicDefineClass(name, b, protectionDomain) : super.defineClass(name, b, 0, b.length, protectionDomain);
+    }
+
+    @Override
+    public EnumSet<ApiType> getApiTypeVisibility() {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        if (appCL != null) {
+            return appCL.getApiTypeVisibility();
+        }
+        if (parent instanceof DeclaredApiAccess) {
+            return ((DeclaredApiAccess) parent).getApiTypeVisibility();
+        }
+        return null;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder(super.toString());
+        sb.append(": Application ClassLoader ");
+        sb.append(classLoaderWeakRef.get());
+        return sb.toString();
+    }
+
+    @Override
+    protected Class<?> loadClass(String className, boolean resolve, boolean onlySearchSelf, boolean returnNull) throws ClassNotFoundException {
+        AppClassLoader appCL = classLoaderWeakRef.get();
+        if (appCL == null) {
+            if (onlySearchSelf) {
```

Comment:
Do we really need to keep trying to have a successful `loadClass` once the application class loader has be GC'ed?